treeherder/webapp/api/infra_serializers.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework import serializers
+
+from django.core.exceptions import ObjectDoesNotExist
+
+from treeherder.model.models import Repository
+
+
+class InfraCompareSerializer(serializers.Serializer):
+    end_time = serializers.DateTimeField(required=False, allow_null=True, default=None)
+    id = serializers.IntegerField()
+    job_type__name = serializers.CharField()
+    result = serializers.CharField()
+    start_time = serializers.DateTimeField(required=False, allow_null=True, default=None)
+
+
+class InfraCompareQuerySerializers(serializers.Serializer):
+    revision = serializers.CharField(required=False, allow_null=False, default=None)
+    project = serializers.CharField()
+    interval = serializers.IntegerField(required=False, allow_null=True, default=None)
+    startday = serializers.DateTimeField(required=False, allow_null=True, default=None)
+    endday = serializers.DateTimeField(required=False, allow_null=True, default=None)
+
+    def validate(self, data):
+        if (
+            data['revision'] is None
+            and data['interval'] is None
+            and (data['startday'] is None or data['endday'] is None)
```

Comment:
do we want one or more of these to be true, or all to be none?

Reply:
If any of these combinations is present (revision or interval or (start_day and day end_day) , it will be sufficient because the the presence of revision will give us the collection of pushes.
interval will give us the commits in that duration. and submit_date and end_date will function similarly. So, we can't keep required=true for any of these.
Repository must be present and I have made it required too

Reply:
> could this be rewritten:
> data['revision'] is None and (data['interval'] is None or (data['startday'] is None or data['endday'] is None))

I think that there will be a few problems with this logic
If only startday or only endday is passed as param, it will get validiated.
But I want it to be false